Python/fileutils.c

```diff
@@ -922,6 +922,95 @@ _Py_GetLocaleEncodingObject(void)
     return str;
 }
 
+#ifdef HAVE_NON_UNICODE_WCHAR_T_REPRESENTATION
+
+/* Check whether current locale uses Unicode as internal wchar_t form. */
+int
+_Py_IsLocaleUnicodeWchar(void)
+{
+    /* Oracle Solaris uses non-Unicode internal wchar_t form for
+       non-Unicode locales and hence needs conversion to UTF first. */
+    char* codeset = nl_langinfo(CODESET);
```

Comment:
While it should not happen, I prefer to be extra safe and handle the NULL case. I suggest to return 0 if nl_langinfo() returns NULL.